packages/foam-core/test/markdown-provider.test.ts

```diff
@@ -72,6 +81,31 @@ describe('Note Title', () => {
     const pageANoteTitle = (graph.getNote('page-d') as Note).title;
     expect(pageANoteTitle).toBe(null);
   });
+
+  it('should give precedence to frontmatter title over other headings', () => {
+    const graph = new NoteGraph();
+    graph.setNote(createNoteFromMarkdown('page-e', pageE, '\n'));
+
+    const pageENoteTitle = (graph.getNote('page-e') as Note).title;
+    expect(pageENoteTitle).toBe('Note Title');
+  });
+});
+
+describe('frontmatter', () => {
+  it('should parse yaml frontmatter', () => {
```

Comment:
we should probably have a case for the empty front matter and the invalid front matter (I don't quite know how to deal with the latter, that test will force us to answer that question :) )